broker-core/src/main/java/io/zeebe/broker/workflow/processor/CatchEventBehavior.java

```diff
@@ -74,33 +86,78 @@ public void subscribeToCatchEvents(
     }
   }
 
-  // CATCH EVENTS
-  private final WorkflowInstanceRecord workflowInstanceRecord = new WorkflowInstanceRecord();
+  public boolean eventOccurred(
```

Comment:
From the method name, it's not clear that we have a bunch of side effects (i.e. modifying the state, writing to the stream), I'd like something that makes it clear you're not just checking if something occurred.

Reply:
Any suggestion?